Source/DafnyDriver/Legacy/SynchronousCliCompilation.cs

```diff
@@ -651,6 +647,18 @@ private record TargetPaths(string Directory, string Filename) {
       var options = dafnyProgram.Options;
       options.Backend.OnPreCompile(dafnyProgram.Reporter, otherFileNames);
 
+      // Process --translation-record options, since translation may need that data to translate correctly.
```

Comment:
Could this new block also move to ExecutionBackend? It seems like it doesn't actual use any of the contents of the program, so it doesn't matter at what point during compilation it's run.

Reply:
It does depend on the set of modules being compiled, and the timing does matter because it needs to capture those names before `ProcessOuterModules` messes them up. :)

But yes it turns out to work well to move it to `ExecutableBackend.Compile`. The only downside is having to ensure overrides like `DafnyExecutableBackend.Compile` also invoke the handling, but that's already a downside for handling like `ProcessOuterModules`. Plus it means the `LibraryBackend` DOESN'T output a translation record by default, and I think that's desirable anyway.